### PR TITLE
[image-picker] Fix typo in iOS code and add handling to JS

### LINKED ISSUE #11285: Typo in ImagePicker, SDK 40
### Summary of Issue 

After testing an app on SDK 40 noticed issue with ImagePicker. Most probably the reason is typo here https://github.com/expo/expo/blob/master/ios/versioned-react-native/ABI40_0_0/Expo/EXImagePicker/ABI40_0_0EXImagePicker/ABI40_0_0EXImagePicker.m#L88

```diff
- ABI40_0_0UM_EXPORT_METHOD_AS(getMediaLibaryPermissionsAsync,
+ ABI40_0_0UM_EXPORT_METHOD_AS(getMediaLibraryPermissionsAsync,
```

### Environment - output of `expo diagnostics` & the platform(s) you're targeting

  Expo CLI 4.0.12 environment info:
    System:
      OS: macOS 10.15.7
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 12.19.1 - /usr/local/bin/node
      Yarn: 1.22.10 - /usr/local/bin/yarn
      npm: 6.14.8 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.9.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.2, DriverKit 20.0, macOS 11.0, tvOS 14.2, watchOS 7.1
      Android SDK:
        API Levels: 28, 29, 30
        Build Tools: 28.0.3, 29.0.2
        System Images: android-24 | Google APIs Intel x86 Atom, android-28 | Intel x86 Atom_64, android-28 | Google APIs Intel x86 Atom, android-29 | Google APIs Intel x86 Atom, android-R | Google APIs Intel x86 Atom
    IDEs:
      Android Studio: 4.0 AI-193.6911.18.40.6626763
      Xcode: 12.2/12B45b - /usr/bin/xcodebuild
    npmPackages:
      expo: next => 40.0.0
      react: 16.13.1 => 16.13.1
      react-dom: 16.13.1 => 16.13.1
      react-native: https://github.com/expo/react-native/archive/sdk-40.0.0.tar.gz => 0.63.2
    npmGlobalPackages:
      expo-cli: 4.0.12
    Expo Workflow: managed

### PR DESCRIPTION
**Pull request by byCedric**
# Why

Fixes #11285

# How

It fixes the typo on the iOS side, and adds handling to the JS side to pick the typo-named method if it exists.

# Test Plan

These two methods should work in the current (including typo) Expo Client:
- `getMediaLibraryPermissionsAsync`
- `requestMediaLibraryPermissionsAsync`

### CODE REVIEW

**Comment by byCedric** [comment]
Jest mocks also needs to be regenerated I think.

**Review by brentvatne — APPROVED** [review]

File: packages/expo-image-picker/src/ImagePicker.ts [hunk 1]
```diff
@@ -76,13 +83,20 @@ export async function requestCameraRollPermissionsAsync(): Promise<MediaLibraryP
...
-  return ExponentImagePicker.requestMediaLibraryPermissionsAsync();
+  return requestMediaLibraryPermissionsAsync();
 }
 
 export async function requestMediaLibraryPermissionsAsync(
   writeOnly: boolean = false
 ): Promise<MediaLibraryPermissionResponse> {
-  return ExponentImagePicker.requestMediaLibraryPermissionsAsync(writeOnly);
+  // due to a typo in iOS, we need to check on the typo too
+  // todo: remove this workaround when a new iOS client is submitted
```
Comment (brentvatne) [inline packages/expo-image-picker/src/ImagePicker.ts:93]:
```suggestion
  // todo: remove this workaround for SDK 41
```

Comment (brentvatne) [reply packages/expo-image-picker/src/ImagePicker.ts:93]: probably safer to keep it until sdk 41

File: packages/expo-image-picker/src/ImagePicker.ts [hunk 2]
```diff
@@ -62,7 +62,14 @@ export async function getCameraRollPermissionsAsync(): Promise<MediaLibraryPermi
 export async function getMediaLibraryPermissionsAsync(
   writeOnly: boolean = false
 ): Promise<MediaLibraryPermissionResponse> {
-  return ExponentImagePicker.getMediaLibraryPermissionsAsync(writeOnly);
+  // due to a typo in iOS, we need to check on the typo too
+  // todo: remove this workaround when a new iOS client is submitted
```
Comment (byCedric) [inline packages/expo-image-picker/src/ImagePicker.ts:66]:
```suggestion
  // todo: remove this workaround for SDK 41
```

**Pull request merged**